plonky2/src/iop/generator.rs

```diff
@@ -106,6 +110,52 @@ pub trait WitnessGenerator<F: Field>: 'static + Send + Sync + Debug {
     /// flag is true, the generator will never be run again, otherwise it will be queued for another
     /// run next time a target in its watch list is populated.
     fn run(&self, witness: &PartitionWitness<F>, out_buffer: &mut GeneratedValues<F>) -> bool;
+
+    fn serialize(&self, dst: &mut Vec<u8>) -> IoResult<()>;
+
+    fn deserialize(src: &mut Buffer) -> IoResult<Self>
+    where
+        Self: Sized;
+}
+
+pub trait AnyWitnessGenerator<F: Field>: WitnessGenerator<F> {
```

Comment:
Why do we need this trait, instead of having `WitnessGeneratorRef` wrap `Box<dyn WitnessGenerator>`?

Reply:
That should suffice indeed. Initially I was puzzled by how to deal here the best way, given that it had to handle both WitnessGenerator and SimpleGenerator, but then I fell back to the "easy" hack around of generators ids and Default impl to get to the SimpleGenerator trait.

Reply:
You're right this one is a leftover from a previous attempt, I can remove it.

Reply:
I applied your comments, in case we opt to keep going for an approach like this one.